zenoh-jni/src/config.rs

```diff
@@ -0,0 +1,133 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use std::{ptr::null, sync::Arc};
+
+use jni::{
+    objects::{JClass, JString},
+    JNIEnv,
+};
+use zenoh::Config;
+
+use crate::errors::Result;
+use crate::{session_error, throw_exception, utils::decode_string};
+
+/// Loads the default configuration, returning a raw pointer to it.
+///
+/// The pointer to the config is expected to be freed later on upon the destruction of the
+/// Kotlin Config instance.
+///
+#[no_mangle]
+#[allow(non_snake_case)]
+pub extern "C" fn Java_io_zenoh_jni_JNIConfig_00024Companion_loadDefaultConfigViaJNI(
```

Comment:
Is this an automatically generated code? The names look a bit strange.

Reply:
No, its not 😅 . It's strange because it's meant to be called from Kotlin through the JNI by a function named `loadDefaultConfigViaJNI` located at io/zenoh/jni/JNIConfig$Companion. It must have this weird looking signature for Java/Kotlin to locate the corresponding native function.